### PR TITLE
Added Compass Calibration programmable operation

### PR DESCRIPTION
**Pull request by sdellava**
This pull request create the programmable operation n. 52

This operation trigger the Compass Calibration procedure.

This operation, activated by an AUX channel, allow the pilot to trigger the compass calibration process in air.

This option eventually solves the issue of calibrating the mag before take off, that in case of presence of large metal masses or if done close to active electronic devices, can result wrong.

### CODE REVIEW

**Comment by DzikuVx** [comment]
@sdellava would would need a Configurator update as well

**Comment by sdellava** [comment]
> @sdellava would would need a Configurator update as well

Yes, I'll submit a second PR asap

**Comment by sdellava** [comment]
Momentary closed to execute additional tests.

**Comment by shota3527** [comment]
It is safe to calibrate compass in air?

**Comment by sdellava** [comment]
Yes it is. I've tested it just one hour ago. But I've find an issue: the calibration process end with a "save setting" procedure that can't be executed while the ARMING_FLAG is TRUE.

The reason is that the function that save the configuration reset the rx link. Therefore it is correct that it can't be execute while armed. 

The effect of this issue is that the system message "**SAVING SETTING**" remain in the OSD until the FC is  ARMING_FLAG is turned to FALSE. Then the save setting procedure complete correctly (in the bottom of the post flight data the "SETTING SAVED" message appears. 

Anyway, on air calibration processe should not be done in a middle of a flight generic, but in a specific calibration flight, and executed only once in a place. So at the moment I'll do no other modifications related to this new feature.

**Comment by sdellava** [comment]
Tested one hour ago. I've found a potential issue: the calibration process end with a "save setting" procedure that can't be executed while the ARMING_FLAG is TRUE.

The reason is that the function that save the configuration reset the rx link. Therefore it is correct that it can't be execute while armed.

The effect of this issue is that the system message "SAVING SETTING" remain in the OSD until the FC is ARMING_FLAG is turned to FALSE. Then the save setting procedure complete correctly (in the bottom of the post flight data the "SETTING SAVED" message appears.

Anyway, on air calibration processe should not be done in a middle of a flight generic, but in a specific calibration flight, and executed only once in a place. So at the moment I'll do no other modifications related to this new feature.

**Comment by sdellava** [comment]
added related configurator PR https://github.com/iNavFlight/inav-configurator/pull/1854

**Comment by sdellava** [comment]
Temporary closed to solve the conflict

**Comment by sdellava** [comment]
Conflic solved

**Comment by sdellava** [comment]
Video of the test here: https://youtu.be/1muitwEfxic

**Comment by sdellava** [comment]
@DzikuVx and team... will this PR will merged or I need to change something?

File: src/main/programming/logic_condition.h [hunk 1]
```diff
@@ -82,8 +82,9 @@ typedef enum {
     LOGIC_CONDITION_DELTA                       = 50,
     LOGIC_CONDITION_APPROX_EQUAL                = 51,
     LOGIC_CONDITION_LED_PIN_PWM                 = 52,
-    LOGIC_CONDITION_DISABLE_GPS_FIX             = 53,
-    LOGIC_CONDITION_LAST                        = 54,
+    LOGIC_CONDITION_RESET_MAG_CALIBRATION       = 53,
+    LOGIC_CONDITION_DISABLE_GPS_FIX             = 54,
```
Comment (DzikuVx) [inline src/main/programming/logic_condition.h]: Your entry has to have a new ID, not reuse existing one. So, `LOGIC_CONDITION_RESET_MAG_CALIBRATION` should be `54` not `53`

**Review by DzikuVx — CHANGES_REQUESTED** [review]
One thing has to be fixed here, but that also means https://github.com/iNavFlight/inav-configurator/pull/1854 has to be updated

**Comment by sdellava** [comment]
Logic condition changed to 54. configurator updated too.

**Pull request merged**